[daily] bicep not found in path

Should be reproducible with our daily container: `azdevcliextacr.azurecr.io/azure-dev:daily`

To reproduce:
1. `azd init` - Choose a template
2. `azd pipeline config --provider azdo` - Create an Azure DevOps pipeline. Say yes to run the pipeline immediately.
3. The pipeline run should succeed.
4. Update the pipeline's yaml file. 
   - Switch `container: mcr.microsoft.com/azure-dev-cli-apps:latest` to `container: azdevcliextacr.azurecr.io/azure-dev:daily`
6. Run the pipeline

Expected:
The pipeline should succeed with the latest `daily` azd image.

Actual:
The pipeline fails with error:
```
ERROR: creating provisioning manager: required external tools are missing:
 - error checking for external tool Bicep CLI: checking Bicep CLI version: cannot find bicep path: stat /home/vsts_azpcontainer/.azure/bin/bicep: no such file or directory
```

```
2022-11-04T20:40:59.5411102Z [command]/usr/bin/az cloud set -n AzureCloud
2022-11-04T20:41:01.4055406Z [command]/usr/bin/az login --service-principal -u *** --password=*** --tenant 72f988bf-86f1-41af-91ab-2d7cd011db47 --allow-no-subscriptions
2022-11-04T20:41:02.8247019Z [command]/usr/bin/az account set --subscription ***
2022-11-04T20:41:03.3811336Z [command]/bin/bash /__w/_temp/azureclitaskscript1667594458394.sh
2022-11-04T20:41:10.5996764Z Error: creating provisioning manager: required external tools are missing:
2022-11-04T20:41:10.5998879Z  - error checking for external tool Bicep CLI: checking Bicep CLI version: cannot find bicep path: stat /home/vsts_azpcontainer/.azure/bin/bicep: no such file or directory
2022-11-04T20:41:10.6118595Z ##[error]Script failed with exit code: 1
2022-11-04T20:41:10.6176663Z [command]/usr/bin/az account clear
2022-11-04T20:41:11.3700034Z ##[section]Finishing: Azure Dev Provision
```

--- a/cli/Dockerfile
+++ b/cli/Dockerfile
@@ -21,7 +21,12 @@ RUN export DEBIAN_FRONTEND=noninteractive \
      # az cli
      && curl -sSL https://aka.ms/InstallAzureCLIDeb | bash \
      # bicep
-     && az bicep install \
+     # Fetch the latest Bicep CLI binary
+     && curl -Lo bicep https://github.com/Azure/bicep/releases/latest/download/bicep-linux-x64 \
+     # Mark it as executable
+     && chmod +x ./bicep \
+     # Add bicep to your PATH (requires admin)
+     && mv ./bicep /usr/local/bin/bicep \
      # nodejs
      && curl -fsSL https://deb.nodesource.com/setup_16.x | bash - \
      && apt-get install -y nodejs \

--- a/cli/azd/pkg/tools/bicep/bicep.go
+++ b/cli/azd/pkg/tools/bicep/bicep.go
@@ -18,8 +18,6 @@ import (
 	"github.com/blang/semver/v4"
 )
 
-const EnvNameAzureConfigDir = "AZURE_CONFIG_DIR"
-
 type BicepCli interface {
 	tools.ExternalTool
 	Build(ctx context.Context, file string) (string, error)
@@ -109,6 +107,7 @@ type contextKey string
 const (
 	bicepContextKey         contextKey = "bicepcli"
 	defaultBicepCommandPath string     = "bicep"
+	envNameAzureConfigDir   string     = "AZURE_CONFIG_DIR"
 )
 
 func GetBicepCli(ctx context.Context) BicepCli {
@@ -157,7 +156,7 @@ func findBicepPath() (*string, error) {
 	commonPaths := []string{}
 
 	// If AZURE_CONFIG_DIR is defined, check there first
-	azureConfigDir := os.Getenv(EnvNameAzureConfigDir)
+	azureConfigDir := os.Getenv(envNameAzureConfigDir)
 	if strings.TrimSpace(azureConfigDir) != "" {
 		commonPaths = append(commonPaths, filepath.Join(azureConfigDir, "bin", bicepFilename))
 	}

--- a/cli/azd/pkg/tools/bicep/bicep.go
+++ b/cli/azd/pkg/tools/bicep/bicep.go
@@ -6,6 +6,7 @@ package bicep
 import (
 	"context"
 	"fmt"
+	"log"
 	"os"
 	"os/user"
 	"path/filepath"
@@ -158,6 +159,7 @@ func findBicepPath() (*string, error) {
 	// If AZURE_CONFIG_DIR is defined, check there first
 	azureConfigDir := os.Getenv(envNameAzureConfigDir)
 	if strings.TrimSpace(azureConfigDir) != "" {
+		log.Printf("Found %s with path '%s'\n", envNameAzureConfigDir, azureConfigDir)
 		commonPaths = append(commonPaths, filepath.Join(azureConfigDir, "bin", bicepFilename))
 	}
 
@@ -176,6 +178,8 @@ func findBicepPath() (*string, error) {
 		if existsErr == nil {
 			return &installPath, nil
 		}
+
+		log.Printf("Bicep CLI not found at path '%s'\n", installPath)
 	}
 
 	return nil, fmt.Errorf("cannot find bicep path: %w", existsErr)

--- a/cli/azd/pkg/tools/bicep/bicep.go
+++ b/cli/azd/pkg/tools/bicep/bicep.go
@@ -159,7 +159,7 @@ func findBicepPath() (*string, error) {
 	// If AZURE_CONFIG_DIR is defined, check there first
 	azureConfigDir := os.Getenv(EnvNameAzureConfigDir)
 	if strings.TrimSpace(azureConfigDir) != "" {
-		commonPaths = append(commonPaths, filepath.Join(azureConfigDir, bicepFilename))
+		commonPaths = append(commonPaths, filepath.Join(azureConfigDir, "bin", bicepFilename))
 	}
 
 	// Check for standalone installation

--- a/cli/azd/pkg/tools/bicep/bicep.go
+++ b/cli/azd/pkg/tools/bicep/bicep.go
@@ -10,12 +10,15 @@ import (
 	"os/user"
 	"path/filepath"
 	"runtime"
+	"strings"
 
 	"github.com/azure/azure-dev/cli/azd/pkg/convert"
 	"github.com/azure/azure-dev/cli/azd/pkg/exec"
 	"github.com/azure/azure-dev/cli/azd/pkg/tools"
 	"github.com/blang/semver/v4"
 )
+
+const EnvNameAzureConfigDir = "AZURE_CONFIG_DIR"
 
 type BicepCli interface {
 	tools.ExternalTool
@@ -139,18 +142,31 @@ func findBicepPath() (*string, error) {
 		return nil, fmt.Errorf("failed getting current user: %w", err)
 	}
 
-	commonPaths := []string{}
-	azureBin := filepath.Join(user.HomeDir, ".azure", "bin")
+	var bicepFilename string
+	var bicepStandalonePath string
 
-	// When installed standalone is typically located in the following locations
-	// When installed with 'az' it is inside the azure bin folder across all OSes
 	if runtime.GOOS == "windows" {
-		commonPaths = append(commonPaths, filepath.Join(user.HomeDir, "AppData\\Local\\Programs\\Bicep CLI\\bicep.exe"))
-		commonPaths = append(commonPaths, filepath.Join(azureBin, "bicep.exe"))
+		bicepFilename = "bicep.exe"
+		bicepStandalonePath = filepath.Join(user.HomeDir, fmt.Sprintf("AppData\\Local\\Programs\\Bicep CLI\\%s", bicepFilename))
 	} else {
-		commonPaths = append(commonPaths, "/usr/local/bin/bicep")
-		commonPaths = append(commonPaths, filepath.Join(azureBin, "bicep"))
+		bicepFilename = "bicep"
+		bicepStandalonePath = fmt.Sprintf("/usr/local/bin/%s", bicepFilename)
 	}
+
+	azureBin := filepath.Join(user.HomeDir, ".azure", "bin")
+	commonPaths := []string{}
+
+	// If AZURE_CONFIG_DIR is defined, check there first
+	azureConfigDir := os.Getenv(EnvNameAzureConfigDir)
+	if strings.TrimSpace(azureConfigDir) != "" {
+		commonPaths = append(commonPaths, filepath.Join(azureConfigDir, bicepFilename))
+	}
+
+	// Check for standalone installation
+	commonPaths = append(commonPaths, bicepStandalonePath)
+
+	// Otherwise look in standard azure bin dir
+	commonPaths = append(commonPaths, filepath.Join(azureBin, bicepFilename))
 
 	var existsErr error
 

